xonsh/ptk_shell/shell.py

```diff
@@ -35,6 +35,12 @@
 from prompt_toolkit.styles import merge_styles, Style
 from prompt_toolkit.styles.pygments import pygments_token_to_classname
 
+try:
+    from prompt_toolkit.clipboard.pyperclip import PyperclipClipboard
+
+    SYS_CLIPBOARD = True
```

Comment:
Maybe `HAVE_SYS_CLIPBOARD`?  Is this version gated? Like could we use something like:

```
if ptk.version >= (3, 2, 1):
  from prompt_toolkit.clipboard.pyperclip import PyperclipClipboard
  HAVE_SYS_CLIPBOARD = True
else:
  HAVE_SYS_CLIPBOARD = False

```

Reply:
Or I guess it needs pyperclip?

Reply:
The current version of ptk is 3.0.8 , xonsh requires ptk v3 already.
This needs pyperclip which is an optional dependency not installed with ptk.

Reply:
right, yeah.  OK.

Reply:
Also, should this be `HAVE_SYS_CLIPBOARD`?

Reply:
Yes will do.

Reply:
updated.